jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -913,14 +930,15 @@ public String(char[] data, int start, int length) {
 				}
 
 				coder = LATIN1;
+			} else if (COMPACT_STRINGS && helpers.canEncodeAsLatin1(data, start, length)) {
+				value = new byte[length];
+				coder = LATIN1;
+
+				compress(data, start, value, 0, length);
 			} else {
-				if (sharingIsAllowed && start == 0 && data.length == length * 2) {
-					value = data;
-				} else {
```

Comment:
Why did you remove this?

Reply:
Since the case is not used.

Reply:
I removed the case from the compressed branch as well, made the other review changes, and removed a few blank lines.